diff -uNr mhash-0.9.9/include/mutils/mglobal.h mhash-0.9.9-hilrom/include/mutils/mglobal.h
--- mhash-0.9.9/include/mutils/mglobal.h	2006-01-09 06:06:16.000000000 +0100
+++ mhash-0.9.9-hilrom/include/mutils/mglobal.h	2008-10-11 15:45:06.000000000 +0200
@@ -68,6 +68,8 @@
 	MHASH_PANAMA		= 70, /* NOT YET IMPLEMENTED */
 	MHASH_SMASH		= 71, /* NOT YET IMPLEMENTED */
 	MHASH_SUBHASH		= 72, /* NOT YET IMPLEMENTED */
+	MHASH_HILROM		= 128,
+	MHASH_HILROMI		= 129,
 #endif
 #if (MHASH_API_TYPE == MHASH_API_FAMILY)
 	MHASH_CRC32		= 0x0001,
@@ -106,6 +108,8 @@
 	MHASH_PANAMA		= 0x0d01, /* NOT YET IMPLEMENTED */
 	MHASH_SMASH		= 0x0e01, /* NOT YET IMPLEMENTED */
 	MHASH_SUBHASH		= 0x0f01, /* NOT YET IMPLEMENTED */
+	MHASH_HILROM		= 0x1001,
+	MHASH_HILROMI		= 0x1002,
 #endif
 } hashid;
 
diff -uNr mhash-0.9.9/lib/hilrom.c mhash-0.9.9-hilrom/lib/hilrom.c
--- mhash-0.9.9/lib/hilrom.c	1970-01-01 01:00:00.000000000 +0100
+++ mhash-0.9.9-hilrom/lib/hilrom.c	2008-10-11 16:09:44.000000000 +0200
@@ -0,0 +1,93 @@
+/*
+ *    Copyright (C) 2008 Christoph Thelen
+ *
+ *    This library is free software; you can redistribute it and/or modify it 
+ *    under the terms of the GNU Library General Public License as published 
+ *    by the Free Software Foundation; either version 2 of the License, or 
+ *    (at your option) any later version.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Library General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Library General Public
+ *    License along with this library; if not, write to the
+ *    Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ *    Boston, MA 02111-1307, USA.
+ */
+
+
+#include "libdefs.h"
+
+#ifdef ENABLE_HILROM
+
+#include "mhash_hilrom.h"
+
+
+void HilRom_Init(struct HilRomContext *context)
+{
+	context->iOffs = 0;
+	context->iIsInverted = 0;
+	context->checksum = 0;
+}
+
+void HilRom_Init_Inv(struct HilRomContext *context)
+{
+	context->iOffs = 0;
+	context->iIsInverted = 1;
+	context->checksum = 0;
+}
+
+
+void HilRom_Update(struct HilRomContext *context, mutils_word8 __const *buf, mutils_word32 len)
+{
+	int iOffset;
+	mutils_word8 __const *pucCnt;
+	mutils_word8 __const *pucEnd;
+	mutils_word32 checksum;
+
+
+	iOffset = context->iOffs << 3;
+	checksum = context->checksum;
+
+	pucCnt = buf;
+	pucEnd = pucCnt + len;
+
+	while(pucCnt<pucEnd)
+	{
+		checksum += *(pucCnt++) << iOffset;
+	
+		/* inc offset */
+		iOffset += 8;
+		iOffset &= 0x1f;
+	}
+
+	context->iOffs = iOffset >> 3;
+	context->checksum = checksum;
+}
+
+void HilRom_Final(struct HilRomContext *context, mutils_word8 *digest)
+{
+	mutils_word32 tmp;
+
+	tmp = context->checksum;
+	if( context->iIsInverted!=0 )
+	{
+		--tmp;
+		tmp ^= 0xffffffff;
+	}
+
+#if defined(WORDS_BIGENDIAN)
+	tmp = mutils_word32swap(tmp);
+#endif
+	if (digest != NULL)
+	{
+		mutils_memcpy(digest, &tmp, sizeof(mutils_word32));	
+	}
+
+	memset(context, 0, sizeof(struct HilRomContext));
+}
+
+
+#endif /* ENABLE_HILROM */
diff -uNr mhash-0.9.9/lib/mhash.c mhash-0.9.9-hilrom/lib/mhash.c
--- mhash-0.9.9/lib/mhash.c	2007-02-21 07:18:46.000000000 +0100
+++ mhash-0.9.9-hilrom/lib/mhash.c	2008-10-11 16:10:41.000000000 +0200
@@ -40,6 +40,7 @@
 #include <mhash_gost.h>
 #include <mhash_whirlpool.h>
 #include <mhash_snefru.h>
+#include <mhash_hilrom.h>
 
 /* 19/03/2000 Changes for better thread handling --nikos 
  * Actually it is thread safe.
@@ -163,6 +164,13 @@
 		    snefru256_update, snefru256_final, snefru256_digest),
 #endif
 
+#if defined(ENABLE_HILROM)
+	MHASH_ENTRY(MHASH_HILROM, 4, 0, sizeof(HILROM_CTX), HilRom_Init, 
+		HilRom_Update, NULL, HilRom_Final),
+	MHASH_ENTRY(MHASH_HILROMI, 4, 0, sizeof(HILROM_CTX), HilRom_Init_Inv,
+		HilRom_Update, NULL, HilRom_Final),
+#endif
+
 	{0}
 };
 
diff -uNr mhash-0.9.9/lib/mhash_hilrom.h mhash-0.9.9-hilrom/lib/mhash_hilrom.h
--- mhash-0.9.9/lib/mhash_hilrom.h	1970-01-01 01:00:00.000000000 +0100
+++ mhash-0.9.9-hilrom/lib/mhash_hilrom.h	2008-10-11 15:14:10.000000000 +0200
@@ -0,0 +1,24 @@
+#if defined(ENABLE_HILROM)
+
+#if !defined(__MHASH_HILROM_H)
+#define __MHASH_HILROM_H
+
+#include "libdefs.h"
+
+struct HilRomContext {
+	int iOffs;			/* current byteoffset in the dword (0..3) */
+	int iIsInverted;		/* !=0 if the checksum is inverted */
+	mutils_word32 checksum;		/* the checksum */
+};
+
+void HilRom_Init(struct HilRomContext *context);
+void HilRom_Init_Inv(struct HilRomContext *context);
+void HilRom_Update(struct HilRomContext *context, mutils_word8 __const *buf, mutils_word32 len);
+void HilRom_Final(struct HilRomContext *context, mutils_word8 *digest);
+
+
+typedef struct HilRomContext HILROM_CTX;
+
+#endif
+
+#endif
