diff -uNr -x '*.svn*' trunk_org/src/helper/command.c trunk/src/helper/command.c
--- trunk_org/src/helper/command.c	2007-12-15 20:02:39.000000000 +0100
+++ trunk/src/helper/command.c	2007-12-15 20:19:43.000000000 +0100
@@ -181,11 +181,11 @@
 	return ERROR_OK;
 }
 
-int parse_line(char *line, char *words[], int max_words)
+int parse_line(const char *line, char *words[], int max_words)
 {
 	int nwords = 0;
-	char *p = line;
-	char *word_start = line;
+	const char *p = line;
+	const char *word_start = line;
 	int inquote = 0;
 
 	while (nwords < max_words - 1)
@@ -201,7 +201,7 @@
 			if (word_start)
 			{
                          int len;
-                            char *word_end=p;
+                            const char *word_end=p;
                             /* This will handle extra whitespace within quotes */
                          while (isspace(*word_start)&&(word_start<word_end))
                                    word_start++;
@@ -252,7 +252,7 @@
 	return nwords;
 }
 
-void command_print(command_context_t *context, char *format, ...)
+void command_print(command_context_t *context, const char *format, ...)
 {
 	va_list ap;
 	char *buffer = NULL;
@@ -339,7 +339,7 @@
 	return ERROR_OK;
 }
 
-int command_run_line(command_context_t *context, char *line)
+int command_run_line(command_context_t *context, const char *line)
 {
 	int nwords;
 	char *words[128] = {0};
diff -uNr -x '*.svn*' trunk_org/src/helper/command.h trunk/src/helper/command.h
--- trunk_org/src/helper/command.h	2007-12-15 20:02:39.000000000 +0100
+++ trunk/src/helper/command.h	2007-12-15 20:24:38.000000000 +0100
@@ -58,8 +58,8 @@
 extern command_context_t* copy_command_context(command_context_t* context);
 extern command_context_t* command_init();
 extern int command_done(command_context_t *context);
-extern void command_print(command_context_t *context, char *format, ...);
-extern int command_run_line(command_context_t *context, char *line);
+extern void command_print(command_context_t *context, const char *format, ...);
+extern int command_run_line(command_context_t *context, const char *line);
 extern int command_run_file(command_context_t *context, FILE *file, enum command_mode mode);
 
 
