diff -uNrw -x '*.svn*' trunk_org/src/helper/command.c trunk/src/helper/command.c
--- trunk_org/src/helper/command.c	2007-12-15 19:45:22.000000000 +0100
+++ trunk/src/helper/command.c	2007-12-15 19:39:44.000000000 +0100
@@ -498,13 +498,51 @@
 {
 	command_context_t* copy_context = malloc(sizeof(command_context_t));
 
-	*copy_context = *context;
+	/* copy context elements */
+	copy_context->mode = context->mode;
+	copy_context->is_copied = 1;
+	copy_context->commands = context->commands;
+	copy_context->current_target = context->current_target;
+	copy_context->echo = context->echo;
+	copy_context->output_handler = context->output_handler;
+	copy_context->output_handler_priv = context->output_handler_priv;
 	
 	return copy_context;
 }
 
+void free_commands(command_t *commands)
+{
+	if( commands->name!=NULL )
+	{
+		free(commands->name);
+	}
+	if( commands->help!=NULL )
+	{
+		free(commands->help);
+	}
+
+	/* free children */
+	if( commands->children!=NULL )
+	{
+		free_commands(commands->children);
+	}
+
+	/* free next */
+	if( commands->next!=NULL )
+	{
+		free_commands(commands->next);
+	}
+
+	/* free commands */
+	free(commands);
+}
+
 int command_done(command_context_t *context)
 {
+	if( context->is_copied==0 && context->commands!=NULL )
+	{
+		free_commands(context->commands);
+	}
 	free(context);
 	
 	return ERROR_OK;
@@ -515,6 +553,7 @@
 	command_context_t* context = malloc(sizeof(command_context_t));
 	
 	context->mode = COMMAND_EXEC;
+	context->is_copied = 0;
 	context->commands = NULL;
 	context->current_target = 0;
 	context->echo = 0;
diff -uNrw -x '*.svn*' trunk_org/src/helper/command.h trunk/src/helper/command.h
--- trunk_org/src/helper/command.h	2007-12-15 19:45:22.000000000 +0100
+++ trunk/src/helper/command.h	2007-11-24 23:09:52.000000000 +0100
@@ -32,6 +32,7 @@
 typedef struct command_context_s
 {
 	enum command_mode mode;
+	int is_copied;
 	struct command_s *commands;
 	int current_target;
 	int echo;
