--- trunk/src/jtag/ft2232.c	2007-11-24 22:35:09.000000000 +0100
+++ /home/Benten/Coding/netx/muhkuh/trunk/plugins/romloader/openocd/openocd/jtag/ft2232.c	2007-12-15 20:32:00.000000000 +0100
@@ -1505,7 +1505,11 @@
 		DEBUG("current latency timer: %i", latency_timer);
 	}
 
-	ftdi_set_bitmode(&ftdic, 0x0b, 2); /* ctx, JTAG I/O mask */
+	if (ftdi_set_bitmode(&ftdic, 0x0b, 2) < 0)	/* ctx, JTAG I/O mask */
+	{
+		ERROR("unable to set bit mode: %s", ftdic.error_str);
+		return ERROR_JTAG_INIT_FAILED;
+	}
 
 	return ERROR_OK;
 }
