diff -uNr org/src/jtag/tcl.c new/src/jtag/tcl.c
--- org/src/jtag/tcl.c	2013-11-11 20:37:38.000000000 +0100
+++ new/src/jtag/tcl.c	2014-01-06 17:46:59.000000000 +0100
@@ -817,12 +829,12 @@
 	if (CMD_ARGC != 0)
 		return ERROR_COMMAND_SYNTAX_ERROR;
 
-	static bool jtag_initialized;
-	if (jtag_initialized) {
-		LOG_INFO("'jtag init' has already been called");
-		return ERROR_OK;
-	}
-	jtag_initialized = true;
+//	static bool jtag_initialized;
+//	if (jtag_initialized) {
+//		LOG_INFO("'jtag init' has already been called");
+//		return ERROR_OK;
+//	}
+//	jtag_initialized = true;
 
 	LOG_DEBUG("Initializing jtag devices...");
 	return jtag_init(CMD_CTX);
